### PR TITLE
Added support for multiple-contract LQDR TVL

### PR DESCRIPTION
**Pull request by jhweintraub**
1. Added support for multiple contracts to be included in TVL not supported by DefiLlama natively
2. Abstracted method to allow for contracts on any supported-network to be queried as long as it following the same event-signature
3. Added support for more fine-grained filters for event-logs, filter Deposits/Withdrawals based on individual-tokens.

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/revest/index.js exports TVL: 
        
 
 
    avalanche                 279.14025999999996
    ethereum                  3.67 M
    polygon                   18.66 k
    fantom                    30.77 k
    
    total                    3.72 M

File: projects/revest/index.js [hunk 1]
```diff
@@ -9,13 +9,35 @@ const {
   transformPolygonAddress,
 } = require("../helper/portedTokens.js");
 
+const { ethers } = require("ethers");
+const { utils } = require("@project-serum/anchor");
```
Comment (g1nt0ki) [inline projects/revest/index.js:13]: I dont think this is used in the code

Comment (jhweintraub) [reply projects/revest/index.js:13]: it does ethers.utils.formatEther(txn.args.amountTokens)

As for the anchor i'm not sure about that, rob added that so I just left it in.

Comment (g1nt0ki) [reply projects/revest/index.js:13]: ethers yes, but I am guessing utils was added by mistake

Comment (RobAnon) [reply projects/revest/index.js:13]: Yes, utils can be removed, meant to reference ethers.utils

**Comment by llamatester** [comment]
Error while running adapter at projects/revest/index.js: 
        
 
 
    
        Please move the following keys into the chain: tvl
    
        We have a new adapter export specification now where tvl and other chain specific information are moved inside chain export.
        For example if your protocol is on ethereum and has tvl and pool2, the export file would look like:
        
            module.exports = {
              methodlogy: '...',
              ethereum: {
                tvl: 
                pool2:
              }
            }

File: projects/revest/index.js [hunk 2]
```diff
@@ -1,81 +1,211 @@
...
+      "fantom",
+      eventLog_addresses["fantom"][x]
+    );
 
-    const balances = {}
-    const tokensAndOwners = []
-    tokens.map(token => owner.map(o => tokensAndOwners.push([token, o])))
-    await sumTokens(balances, tokensAndOwners, block, chain, transformAddress, { resolveCrv: true, resolveLP: true, resolveYearn: true })
-    return balances
+    //TBH i'm not sure this is the intended solution but i'm rolling with it anyways
```
Comment (g1nt0ki) [inline projects/revest/index.js:74]: yeah, this part worries me a bit like as these logs would only over time and become heavy and break the adapter. I looked into the code but I dont see any other way to compute, but also I am a noob, is there a less intensive way of doing this, maybe via a graph query somewhere?

(Thanks for fixing the merge conflict)

Comment (jhweintraub) [reply projects/revest/index.js:74]: you could probably use the graph to compute these things but it also makes the code but more unruly and will take longer. That's a Rob question.

Comment (jhweintraub) [reply projects/revest/index.js:74]: @g1nt0ki I have rolled back the remote-branch to the last commit before the merge so we can coordinate on resolving the conflicts going forward. My bad

**Comment by g1nt0ki** [comment]
@jhweintraub closing this for now, please put up a new PR when graph is ready

**Pull request closed without merging**